### PR TITLE
Test installation of all provider packages

### PR DESCRIPTION
**Pull request by potiuk**
---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: scripts/ci/in_container/run_test_package_installation.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    echo "Uninstalled ${PACKAGE_NAME}"
+    echo "-----------------------------------------------------------------------------------"
+    AIRFLOW_VERSION=$(pip freeze | grep "apache-airflow==" | sed "s/apache-airflow==//")
+    echo "Airflow version after installation ${AIRFLOW_VERSION}"
+    if [[ ${AIRFLOW_VERSION} != "${INSTALL_AIRFLOW_VERSION}" ]]; then
+        echo
+        echo "ERROR! Installing ${PACKAGE_FILE} caused Airflow to upgrade to ${AIRFLOW_VERSION}"
+        echo
+        echo "Please fix dependencies in the package"
+        exit 1
```
Comment (turbaszek) [inline scripts/ci/in_container/run_test_package_installation.sh:61]: Nice!

**Review by turbaszek — APPROVED** [review]

**Comment by turbaszek** [comment]
Just out of curiosity, this PR has no JIRA ticket and no Github Issue. Is it ok? How will we include it in the changelog? @potiuk @ashb

**Comment by potiuk** [comment]
We discussed that not all PRs need to have issues. PR# is more than enough to identify changes and we are going to use PR# in changelog from now on.

**Comment by potiuk** [comment]
@turbaszek - see 
https://lists.apache.org/thread.html/r19014fac5aa2aa15a86890aa438940da263e430cb535ca451ef451a4%40%3Cdev.airflow.apache.org%3E

**Comment by mik-laj** [comment]
I started reviewwing.

File: scripts/ci/ci_prepare_backport_packages.sh [hunk 2]
```diff
@@ -20,14 +20,47 @@ export PYTHON_VERSION=${PYTHON_VERSION:-3.6}
 # shellcheck source=scripts/ci/_script_init.sh
 . "$( dirname "${BASH_SOURCE[0]}" )/_script_init.sh"
 
+LIST_OF_DIRS_FILE=$(mktemp)
+
+cd "${MY_DIR}/../../airflow/providers" || exit 1
+
+find . -type d | sed 's/.\///' | sed 's/\//\./' | grep -E 'hooks|operators|sensors|secrets' \
```
Comment (mik-laj) [inline scripts/ci/ci_prepare_backport_packages.sh]: This command gives inconsistent results regarding the dot and slash. Is this expected?
``` 
redis.sensors
redis.operators
redis.hooks
google.ads/operators
google.ads/hooks
google.marketing_platform/sensors
google.marketing_platform/operators
google.marketing_platform/hooks
```

Comment (mik-laj) [inline scripts/ci/ci_prepare_backport_packages.sh]:
```suggestion
find . -type d | sed 's/.\///; s/\//\./' | grep -E 'hooks|operators|sensors|secrets' \
```

Comment (potiuk) [reply scripts/ci/ci_prepare_backport_packages.sh]: Fixed. Was not intentional but it worked anyway ;)

Comment (potiuk) [reply scripts/ci/ci_prepare_backport_packages.sh]: Yep

**Comment by potiuk** [comment]
All Fixed and rebased @mik-laj

**Review by mik-laj — APPROVED** [review]

**Review by mik-laj — APPROVED** [review]

**Pull request merged**